### PR TITLE
docs(v3-migration-doc): add instruction for logs collector migration without data loss

### PR DESCRIPTION
**Pull request by sumo-drosiek**
add instruction for logs collector migration without data loss

### Checklist

- [ ] Changelog updated or skip changelog label added
- [ ] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

File: docs/v3-migration-doc.md [hunk 1]
```diff
@@ -238,6 +245,51 @@ fluent-bit:
 
 After the upgrade, once OpenTelemetry Collector is running, you can disable Fluent Bit again and proceed without any data loss.
 
+##### Migration without data duplication and data loss (advanced)
+
+If you don't want to loose data and to duplicate them, you can go wih more complex solution.
```
Comment (sumo-drosiek) [inline docs/v3-migration-doc.md:250]:
```suggestion
If you don't want to lose data and to duplicate them, you can go wih more complex solution.
```

File: docs/v3-migration-doc.md [hunk 2]
```diff
@@ -238,6 +245,51 @@ fluent-bit:
 
 After the upgrade, once OpenTelemetry Collector is running, you can disable Fluent Bit again and proceed without any data loss.
 
+##### Migration without data duplication and data loss (advanced)
+
+If you don't want to lose data and to duplicate them, you can go wih more complex solution.
+The idea is to have two separated group of nodes. One for Fluent Bit and one for OpenTelemetry Collector.
+
+Node group for FLuent Bit should contains all and only existing nodes. The second group of nodes is dedicated to all new pods.
```
Comment (kkujawa-sumo) [inline docs/v3-migration-doc.md:253]: typo: FLuent -> Fluent

**Review by kkujawa-sumo — APPROVED** [review]

File: docs/v3-migration-doc.md (hunk 2 above)
Comment (sumo-drosiek) [reply docs/v3-migration-doc.md:253]:
```suggestion
Node group for Fluent Bit should contains all and only existing nodes. The second group of nodes is dedicated to all new pods.
```

File: docs/v3-migration-doc.md [hunk 3]
```diff
@@ -238,6 +245,51 @@ fluent-bit:
 
 After the upgrade, once OpenTelemetry Collector is running, you can disable Fluent Bit again and proceed without any data loss.
 
+##### Migration without data duplication and data loss (advanced)
+
+If you don't want to lose data and to duplicate them, you can go wih more complex solution.
```
Comment (swiatekm-sumo) [inline docs/v3-migration-doc.md:250]:
```suggestion
If you want to migrate without losing data or ingesting duplicates, you can go with a more complex solution.
```

File: docs/v3-migration-doc.md [hunk 4]
```diff
@@ -238,6 +245,51 @@ fluent-bit:
...
+The idea is to have two separated group of nodes. One for Fluent Bit and one for OpenTelemetry Collector.
```
Comment (swiatekm-sumo) [inline docs/v3-migration-doc.md:251]:
```suggestion
The idea is to have two separated groups of nodes. One for Fluent Bit and one for OpenTelemetry Collector.
```

File: docs/v3-migration-doc.md [hunk 5]
```diff
@@ -238,6 +245,51 @@ fluent-bit:
...
+
+Node group for Fluent Bit should contains all and only existing nodes. The second group of nodes is dedicated to all new pods.
```
Comment (swiatekm-sumo) [inline docs/v3-migration-doc.md:253]:
```suggestion
The node group for Fluent Bit should contain all existing nodes. The second group of nodes is dedicated to all new pods.
```

File: docs/v3-migration-doc.md [hunk 6]
```diff
@@ -238,6 +245,51 @@ fluent-bit:
...
+
+Let's consider an example for that.
+We added `workerGroup: old-worker-group` label to all existing nodes, and then apply the following configuration:
```
Comment (swiatekm-sumo) [inline docs/v3-migration-doc.md:256]:
```suggestion
We added `workerGroup: old-worker-group` label to all existing nodes, and then applied the following configuration:
```

File: docs/v3-migration-doc.md [hunk 7]
```diff
@@ -238,6 +245,51 @@ fluent-bit:
...
+          nodeSelectorTerms:
+          - matchExpressions:
+            ## run OpenTelemetry Logs Collector on all nodes which do not match label: workerGroup=old-worker-group
+            - key: workerGroup
+              operator: NotIn
+              values:
+              - old-worker-group
+```
+
+After upgrading the collection, we drained all nodes with `workerGroup: old-worker-group`,
```
Comment (swiatekm-sumo) [inline docs/v3-migration-doc.md:282]:
```suggestion
After upgrading the Helm Chart, we drained all nodes with `workerGroup: old-worker-group`,
```

**Review by swiatekm-sumo — APPROVED** [review]
Typos and phrasing

**Pull request merged**